src/components/Post/ActionsBar/ReportButton/index.tsx

```diff
@@ -153,33 +161,70 @@ const ReportButton: FC<IReportButtonProps> = (props) => {
 
 		setLoading(false);
 	};
-
+	const handleDelete = async () => {
+		setLoading(true);
```

Comment:
early return : `if(!allowed_roles?.includes('moderator') || isNaN(Number(postId)) return;`

Reply:
this should be done before setLoading to true else it'll be loading forever